include/engine/engine.hpp

```diff
@@ -71,7 +72,7 @@ class Engine final
     Status Tile(const api::TileParameters &parameters, std::string &result) const;
 
   private:
-    std::unique_ptr<EngineLock> lock;
+    mutable std::unique_ptr<storage::SharedBarriers> lock;
```

Comment:
Semantically the `mutable` has to be on the barrier and not the pointer, but I don't think we need the mutable here at all. The reason for this is a `const Engine` is not able to modify the pointer but _is_ able to modify the barrier through the `const` pointer.

Also the ctor doing a make_unique can already be done in the ctor initializer list as in `Engine() : lock{ shared_mem ? make_unique() : null } { }`. Then even the unique_ptr could be `const`.